src/dotnet-svcutil/lib/src/CodeDomFixup/CodeDomHelpers.cs

```diff
@@ -0,0 +1,271 @@
+        {
+            CodeMemberMethod m = new CodeMemberMethod();
+            m.Name = method.Name;
+            m.ImplementationTypes.Add(ifaceType);
+            m.Attributes = MemberAttributes.Public | MemberAttributes.Final;
+            foreach (CodeParameterDeclarationExpression parameter in method.Parameters)
+            {
+                CodeParameterDeclarationExpression newParam = new CodeParameterDeclarationExpression(parameter.Type, parameter.Name);
+                newParam.Direction = parameter.Direction;
+                m.Parameters.Add(newParam);
+            }
+            m.ReturnType = method.ReturnType;
+            return m;
+        }
+
+        static string GetMethodNameBase(CodeMemberMethod method)
+        {
+            string methodName = method.Name;
+            if (IsBeginMethod(method))
+            {
+                methodName = StringHelpers.StripPrefix(methodName, "On");
+                methodName = StringHelpers.StripPrefix(methodName, "Begin");
+            }
+            else if (IsEndMethod(method))
+            {
+                methodName = StringHelpers.StripPrefix(methodName, "On");
+            }
+            return methodName;
+        }
+
+        internal static bool IsTaskAsyncMethod(CodeMemberMethod method)
+        {
+            return (method.Name.EndsWith("Async", StringComparison.Ordinal) &&
+                (MatchType(method.ReturnType, new CodeTypeReference(typeof(Task<>)), false, true) ||
+                MatchType<Task>(method.ReturnType)));
+        }
+
+        internal static bool IsBeginMethod(CodeMemberMethod method)
+        {
+            int paramCount = method.Parameters.Count;
+            return (method.Name.StartsWith("Begin", StringComparison.Ordinal) || method.Name.StartsWith("OnBegin", StringComparison.Ordinal)) &&
+                   paramCount >= 2 &&
+                   MatchType<AsyncCallback>(method.Parameters[paramCount - 2].Type) &&
+                   MatchType<object>(method.Parameters[paramCount - 1].Type) &&
+                   MatchType<IAsyncResult>(method.ReturnType);
+        }
+
+        internal static bool IsEndMethod(CodeMemberMethod method)
+        {
+            int paramCount = method.Parameters.Count;
+            return (method.Name.StartsWith("End", StringComparison.Ordinal) || method.Name.StartsWith("OnEnd", StringComparison.Ordinal)) &&
+                   paramCount >= 1 &&
+                   (MatchType<IAsyncResult>(method.Parameters[paramCount - 1].Type) ||
+                   MatchType<IAsyncResult>(method.Parameters[0].Type));
+        }
+
+        internal static bool IsOnewayMethod(CodeMemberMethod method)
+        {
+            CodeAttributeDeclaration operationContract = FindAttribute<OperationContractAttribute>(method.CustomAttributes);
+            foreach (CodeAttributeArgument arg in operationContract.Arguments)
```

Comment:
I'm trying to understand the purpose of this code. It looks like it's looking at an already generated contract from a loaded assembly, but I thought this project was limited to generating the code from the WSDL. This looks like it's used for doing things the other way around?

Reply:
This code was not being used. I removed it.